doc/design/session.md

```diff
@@ -0,0 +1,62 @@
+# Design Doc: Session
+
+## Abstract
+
+This design doc proposes to have an object called *Session* which
+encapsulates the environment in which the computation graph is
+executed.
+
+## Background
+
+A computation graph is executed in an environment which contains the
+[scope](./scope.md) and other states. PaddlePaddle used to only have
```

Comment:
Sorry, what do you mean by "other states"?

Reply:
scope, device, context etc.

Reply:
I have the same question with @typhoonzero , @jacquesqiao  do you mean the *Session* contains runtime resources? 
e.g. already allocated memory in scope, occupied device, etc?

Reply:
Sorry for the late reply. @dzhwinter The environment contains runtime resources. The session  is a "owner" of these runtime resources.